src/index.ts

```diff
@@ -153,11 +153,15 @@ const operatingSystemRules: OperatingSystemRule[] = [
   ['Search Bot', SEARCHBOT_OS_REGEX],
 ];
 
-export function detect(): BrowserInfo | BotInfo | NodeInfo | null {
+export function detect(userAgent: string | null = null): BrowserInfo | BotInfo | NodeInfo | null {
```

Comment:
How do you feel about using the `?:` argument form instead to have the `userAgent` argument receive an `undefined` value if not specified?  Personally, I feel like this is probably more in line with what happens when someone calls the function with zero args.

e.g.

```ts
detect(userAgent?: string): ...
```